taskcluster/ci/maybe-nightly/kind.yml

```
@@ -0,0 +1,47 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - ac_taskgraph.transforms.maybe_nightly:transforms
+    - taskgraph.transforms.job:transforms
+    - taskgraph.transforms.task:transforms
+
+task-defaults:
+    run-on-tasks-for: [github-push]
+    run-on-git-branches: ['main$']
```

Bug: Let's safeguard the regex

```suggestion
    run-on-git-branches: ['^(refs/heads/)?main$']
```

Otherwise branches named something like `new-domain` will target this job.

---

I don't think so.  It uses `re.match` which looks for a match at the beginning of the string.

---

Oh yeah, I had forgotten I put `re.match()` back in https://github.com/taskcluster/taskgraph/commit/026477cd69ce194285ddc3234ffbea9d4649e722#diff-a5b995d44a0f435d183a38a9b26dfb35bcc0c204428d2cce539daf05297bea33R76. 

LGTM then!